src/Bicep.Core/SourceGraph/BicepSourceFile.cs

```diff
@@ -110,6 +110,31 @@ public ResultWithDiagnosticBuilder<AuxiliaryFile> TryLoadAuxiliaryFile(RelativeP
                 });
         }
 
+        public ResultWithDiagnosticBuilder<IEnumerable<AuxiliaryFile>> TryGetAuxiliaryFiles(RelativePath relativePath, string searchPattern = "")
+        {
+            if (this.FileHandle is DummyFileHandle)
+            {
+                // This is only invoked when building SnippetCache. The error is swallowed.
+                return new(x => x.ErrorOccurredReadingFile("Cannot load auxiliary file from dummy file handle"));
+            }
+
+            var directoryHandle = this.FileHandle.GetParent().GetDirectory(relativePath);
+            if (!directoryHandle.Exists())
+            {
+                return new (x => x.ErrorOccuredBrowsingDirectory($"Directory {relativePath} does not exist or additional permissions are necessary to access it"));
```

Comment:
In general, we want to put any localizable text in the DiagnosticsBuilder class. Could we instead make this a more specific error like the following:
```csharp
return new(x => x.DirectoryDoesNotExist(relativePath))
```